core/cog-launcher.c

```diff
@@ -69,6 +70,25 @@ on_action_open (G_GNUC_UNUSED GAction *action,
                               g_variant_get_string (param, NULL));
 }
 
+static gboolean
+decidePermissionRequest (WebKitWebView *webView, WebKitPermissionRequest *request, CogLauncher *launcher)
+{
+    if (!g_strcmp0 (launcher->permissions, "all"))
+        webkit_permission_request_allow (request);
+    else
+        webkit_permission_request_deny (request);
+
+    return TRUE;
+}
+
+static void
+on_new_webview (CogShell *shell, GParamSpec * arg G_GNUC_UNUSED, CogLauncher *launcher)
```

Comment:
This is handling the signal `notify::web-view`, so I would call this function `on_notify_web_view` instead.